e2e/transaction_test.go

```diff
@@ -205,8 +205,14 @@ func TestEthTransfer(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 			defer cancel()
 			receipt, err := srv.WaitForReceipt(ctx, txnHash)
-			assert.NoError(t, err)
-			assert.NotNil(t, receipt)
+
+			if testCase.shouldSuccess {
+				assert.NoError(t, err)
+				assert.NotNil(t, receipt)
```

Comment:
You can merge this with the next if:
```suggestion
			if testCase.shouldSuccess {
				assert.NoError(t, err)
				assert.NotNil(t, receipt)
				fee = new(big.Int).Mul(
					big.NewInt(int64(receipt.GasUsed)),
					big.NewInt(int64(txnObject.GasPrice)),
				)
```

Reply:
👍 https://github.com/0xPolygon/polygon-sdk/pull/103/commits/397ef22344def383649dce484f776a463e4cc428